crates/storage/provider/src/providers/database/provider.rs

```
@@ -4066,7 +4066,9 @@ impl<TX: DbTxMut + DbTx> WalletStateSyncWriter for DatabaseProvider<TX> {
             .map(|(_, record)| record);
 
         if let Some(mut wallet_state_sync_record) = wallet_state_sync_record {
-            wallet_state_sync_record.append_data_chunks(data);
+            for (block, data_chunk) in data {
+                wallet_state_sync_record.append_data_with_block(data_chunk, block);
+            }
             self.tx.put::<tables::WalletStateSyncs>(peer_id, wallet_state_sync_record)?;
         }
         Ok(())
```

Noticed that we do not return an error here if we dont the `wallet_state_sync_record` is `None` and we silently succeed. Is this fine?

---

This is the same pattern that has been applied elsewhere in the other methods on db queries where `seek_exact` has been used. The problem here is that all the db interfaces either return system errors or just empty - both on success and not found. I have just followed this pattern. (cf. pub type ProviderResult<Ok> = Result<Ok, ProviderError>;) If we were to change this now, this might break the api design and start feeling weird. The idea behind it is - you search if you find your thing, you modify it or return it (in many methods there is Option<T> being returned btw.) and you dont fail on non-found items. Only in system errors.